lib/Driver/ToolChains.cpp

```diff
@@ -1047,6 +1047,12 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
     break;
   case LinkKind::DynamicLibrary:
     Arguments.push_back("-shared");
+    if (getTriple().getOS() == llvm::Triple::Linux) {
+      if (getTriple().getSubArch() == llvm::Triple::SubArchType::ARMSubArch_v7 ||
+          getTriple().getSubArch() == llvm::Triple::SubArchType::ARMSubArch_v6) {
+        Arguments.push_back("-Wl,-Bsymbolic");
+      }
+    }
```

Comment:
Another nit-pick: Above in `lib/Driver/Driver.cpp` you check for `triple.getArch() == llvm::Triple::arm` before checking whether `getSubArch()` is equal to `ARMSubArch_v7` or `ARMSubArch_v6`. Is the check for `arm` redundant in `lib/Driver/Driver.cpp`? If not, should it be added here? I think either is fine, as long as the checks are consistent.